Frequent Random NSIS Install CI Failures

eg. https://github.com/freeorion/freeorion/actions/runs/3400026562/jobs/5654170357

Re-running the test builds will often result in success, and other test builds usually work without problem.
````
Run irm get.scoop.sh -outfile 'install.ps1'
  irm get.scoop.sh -outfile 'install.ps1'
  .\install.ps1 -RunAsAdmin
  scoop bucket add extras
  scoop install nsis
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
Initializing...
Downloading...
Extracting...
Creating shim...
Adding ~\scoop\shims to your path.
Scoop was installed successfully!
Type 'scoop help' for instructions.
Checking repo... OK
The extras bucket was added successfully.
Installing 'nsis' (3.08) [64bit] from extras bucket
The SSL connection could not be established, see inner exception.
URL https://downloads.sourceforge.net/project/nsis/NSIS%203/3.08/nsis-3.08.zip is not valid
Error: Process completed with exit code 1.
````

--- a/.github/workflows/build-windows-msvs.yml
+++ b/.github/workflows/build-windows-msvs.yml
@@ -47,8 +47,15 @@ jobs:
         run: |
           cd msvc2022
           msbuild FreeOrion.sln /maxcpucount /property:BuildInParallel=true /property:CL_MPCount=2 /property:PlatformToolset=v142 /property:Configuration=Release /verbosity:minimal
+      - name: Cache NSIS
+        uses: actions/cache@v3
+        with:
+          key: scoop-cache
+          path: scoop_cache
       - name: Install NSIS
         if: true # only for weekly-test-build push
+        env:
+          SCOOP_CACHE: scoop_cache
         run: |
           irm get.scoop.sh -outfile 'install.ps1'
           .\install.ps1 -RunAsAdmin

